h2o-core/src/main/java/water/util/CompressionFactory.java

```diff
@@ -0,0 +1,57 @@
+package water.util;
+
+import water.Iced;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.zip.GZIPOutputStream;
+
+public class CompressionFactory extends Iced<CompressionFactory> {
+
+  private String _name;
+
+  private CompressionFactory(String name) {
+    _name = name;
+  }
+
+  OutputStream wrapOutputStream(OutputStream os) throws IOException {
+    final String n = _name.toLowerCase();
+    switch (n) {
```

Comment:
I'd suggest creating enumeration instead of string passing. The validation of an enum is also easier. In the handler, let's just use enum's `valueOf()` method to convert user-provided string to ENUM on the backend.

Reply:
It supports more than just an enumerated set of values, you can specify any class name and it will try to use that. This way users are not limited to what we've baked in.

wdyt?